api.planx.uk/gis/digitalLand.js

```diff
@@ -185,7 +190,23 @@ async function go(localAuthority, geom, extras) {
     }
   }
 
-  return { url: url, constraints: formattedResult };
+  // --- METADATA ---
+  // additionally fetch metadata from Digital Land's "dataset" endpoint for extra context
+  let metadata = {};
+  const urls = activeDatasets.map((dataset) => `https://www.planning.data.gov.uk/dataset/${dataset}.json`);
+  await Promise.all(urls.map(url => 
+    fetch(url)
+      .then(response => response.json())
+      .catch(error => console.log(error))
+  )).then((responses) => {
+    responses.forEach((response) => {
+      // get the planx variable that corresponds to this 'dataset', should never be null because we only requested known datasets
+      const key = Object.keys(baseSchema).find((key) => baseSchema[key]["digital-land-datasets"].includes(response.dataset));
+      metadata[key] = response;
+    });
+  }).catch(error => console.log(error));
+
+  return { url: url, constraints: formattedResult, metadata: metadata };
```

Comment:
Extracting this new code out into its own function would be nice but that could also be a task for a full refactor of this large function.

Reply:
Yes, good shout - this file is definitely a near future candiate for js > ts conversion and breaking down into smaller methods. I'll come back to in a followup for the sake of getting the frontend PR ready for Ian first.